### PR TITLE
Add executor field to the DB and parameter to the operators

### PR DESCRIPTION
**Pull request by ferruzzi**
# Please do not merge before 2.9.0b2 is cut!

Resumbitting https://github.com/apache/airflow/pull/38054

@ferruzzi accidentally merged it when it required a few other PRs to get merged first.  

# Please read the thread on that previous PR before merging this 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by ferruzzi** [comment]
Previously had approval from @potiuk and @eladkal , safe to merge after https://github.com/apache/airflow/pull/38446 "and one other PR" See [here](https://github.com/apache/airflow/pull/38054#issuecomment-2018507597) for details.   

## Don't be a ferruzzi! :P

**Review by potiuk — APPROVED** [review]

File: airflow/migrations/versions/0139_2_10_0_add_new_executor_field_to_db.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#
```
Comment (potiuk) [inline airflow/migrations/versions/0139_2_10_0_add_new_executor_field_to_db.py]: Please do not merge before 2.0.9b2 is cut!

**Comment by potiuk** [comment]
Left an unresolved conversation to avoid accidental merge

**Review by uranusjr — APPROVED** [review]

**Comment by o-nikolas** [comment]
@potiuk Looks like 2.9.0b2 is out now. Am I all clear to merge this now?

**Comment by potiuk** [comment]
@ephraimbuddy -> are you going to just cherry-pick for RC1? No more moving the branch :)?

**Comment by ephraimbuddy** [comment]
> @ephraimbuddy -> are you going to just cherry-pick for RC1? No more moving the branch :)?

yes for the RC. We moved the branch for beta2 so RC1 is going to be cherrypicked

**Review by ephraimbuddy — APPROVED** [review]

**Comment by potiuk** [comment]
go ahead @o-nikolas

**Pull request merged**